src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -703,6 +703,8 @@ void SmartAI::OnCharmed(bool /*isNew*/)
     }
 
     GetScript()->ProcessEventsFor(SMART_EVENT_CHARMED, nullptr, 0, 0, charmed);
+
+    UnitAI::OnCharmed(isNew);
```

Comment:
did you check CreatureAI::OnCharmed() too ? any idea about which one of the 2 should be called (as CreatureAI is the base class of SmartAI) and what kind of issues the 2 might create ?

Reply:
By looking at https://github.com/TrinityCore/TrinityCore/blob/e33384d4ddbd4477f8cc5cae6e0679ab914595a5/src/server/game/AI/CoreAI/UnitAI.h#L148-L152 it seems UnitAI::OnCharmed(isNew) is does the same as what the comment says, so calling ScheduleAIChange() , without having to trigger anything more that CreatureAI might do.
So this change should be ok.